Sprint 23

h1node opened the pull request:
- app icon
 - Playlist screen
 - list of tracks in the playlist
 - menu with additional features
 - Editing a playlist

h1node committed:
Sprint 23
 - app icon
 - Playlist screen
 - list of tracks in the playlist
 - menu with additional features
 - Editing a playlist

ArturNurtdinov commented on .idea/deploymentTargetSelector.xml:
Всю папку .idea лучше сразу добавлять в gitignore - в ней хранятся локальные настройки разработчика, специфичные для проекта

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/domain/usecase/DeletePlaylistUseCase.kt:
```diff
@@ -0,0 +1,13 @@
+import com.playlistmaker.domain.repositories.PlaylistRepository
+
+
+class DeletePlaylistUseCase(
+    private val repository: PlaylistRepository
+) {
```
Юзкейсы тоже стоит разделять на интерфейс и реализацию для поддержания гибкости архитектуры

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/presentation/ui/media/fragments/EditPlaylistFragment.kt:
```diff
@@ -0,0 +1,72 @@
+        viewModel.onNameChanged(playlist.name)
+        viewModel.onDescriptionChanged(playlist.description)
+        playlist.coverUri?.let { filename ->
+            val file = File(requireContext().filesDir, filename)
+            if (file.exists()) {
+                viewModel.onCoverUriChanged(Uri.fromFile(file))
+            }
```
Работу с файлами лучше реализовывать исключительно в репозитории, на ui можно пользоваться классом Uri для доступа к картинке

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/presentation/ui/media/fragments/PlaylistDetailsFragment.kt:
```diff
@@ -0,0 +1,287 @@
+        menuSheetBehavior.isHideable = true
+
+        val overlay = View(requireContext()).apply {
+            setBackgroundColor(0x99000000.toInt())
```
Цвета лучше не хардкодить таким образом, эту тень можно прямо в вёрстке реализовать, чтобы не вызывать добавление вью в иерархию после инфлейта всего экрана

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/presentation/ui/media/fragments/PlaylistDetailsFragment.kt:
```diff
@@ -0,0 +1,287 @@
+        startActivity(Intent.createChooser(intent, getString(R.string.share_playlist)))
+    }
+
+    private fun buildShareText(playlist: Playlist, tracks: List<Music>): String {
```
Функцию со сборкой текста для отправки плейлиста лучше перенести в репозиторий, слой `presentation` должен отвечать только за отображение данных и обработку нажатий

ArturNurtdinov reviewed (commented):
1. Диалог добавления в плейлист на экране аудиоплеера открывается на весь экран. По макетам диалог открывается не на весь экран:

<img width="355" height="791" alt="image" src="https://github.com/user-attachments/assets/5ea618a6-a109-44eb-b2f7-1e57c49d8210" />

Макет:

<img width="359" height="795" alt="image" src="https://github.com/user-attachments/assets/303a3e50-837a-4a14-82a3-7292d1443bf2" />

2. На экране плейлиста присутствует лишний заголовок "PlaylistMaker", обложка имеет лишние отступы:
<img width="356" height="481" alt="image" src="https://github.com/user-attachments/assets/8faeb2af-d5ab-4e3d-bab2-264bb012295b" />

Макет:

<img width="440" height="639" alt="image" src="https://github.com/user-attachments/assets/f8aefad7-8d19-4226-b62a-285f57c3ba1b" />

3. Высота списка треков на экране плейлиста не соответствует макетам:
<img width="333" height="397" alt="image" src="https://github.com/user-attachments/assets/54e088c9-7473-4da7-a0ad-7a84a6857dfe" />

Макет:

<img width="430" height="528" alt="image" src="https://github.com/user-attachments/assets/c81e2319-50b6-42db-9b79-641ca1ed8dbb" />

4. Нижняя панель навигации отображается на экране информации о плейлисте (скриншот выше)

6. Удаление плейлистов не работает. Плейлист продолжает отображаться на экране медиатеки и сохраняется возможность добавлять в него треки - https://disk.yandex.ru/i/8EI1hgSrTnB4IA

h1node committed:
Sprint 23
 - fix layout
 - fix delete playlist
 - комментарии разберу позже

ArturNurtdinov reviewed (commented):
1. Диалог добавления в плейлист на экране аудиоплеера открывается без затемнения фона и закругления углов:
<img width="379" height="820" alt="image" src="https://github.com/user-attachments/assets/9191ab36-f154-4507-b147-42a33f50c3f2" />

Макет:
<img width="364" height="747" alt="image" src="https://github.com/user-attachments/assets/110bbd6e-224a-4f88-9693-89e999ddd002" />

2. Высота списка треков на экране плейлиста не соответствует макетам:
<img width="332" height="384" alt="image" src="https://github.com/user-attachments/assets/3006637d-c1dd-4b32-b751-71bf080692c1" />

Макет:
<img width="458" height="557" alt="image" src="https://github.com/user-attachments/assets/11307474-54e5-4157-b729-45a84db21d1e" />

3. Закругление углов обложки плейлиста в списке плейлистов не соответствует макетам (нижние два угла не закруглены):
<img width="289" height="319" alt="image" src="https://github.com/user-attachments/assets/3d3c0cb9-7fc4-4e2d-a520-3141ca784cba" />

Макет:
<img width="371" height="428" alt="image" src="https://github.com/user-attachments/assets/32bfcd71-86ee-4a06-9052-6f6a0b4041d6" />

h1node committed:
Sprint 23
 - fix layout
 - fix Диалог добавления в плейлист
 - fix Список треков(плейлист) отсортирован
